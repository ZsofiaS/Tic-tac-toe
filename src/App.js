import React from 'react';
import Button from './Button';

function App() {
  return (
    <div>
      <h1>Hello</h1>
      <h2>Start editing</h2>
      <Button onClick={() => console.log('clicked!')}>I am a button</Button>
      <Button primary>I am a primary button</Button>
    </div>
  );
}

export default App;
