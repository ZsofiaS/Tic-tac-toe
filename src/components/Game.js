import React, { useState } from 'react';
import { Board } from './Board';

const Game = () => {
  const [currentPlayer, setCurrentPlayer] = useState('X');
  const [winner, setWinner] = useState('');

  const updatePlayer = () => {
    currentPlayer === 'X' ? setCurrentPlayer('O') : setCurrentPlayer('X');
  };

  const declareWinner = () => {
    currentPlayer === 'X' ? setWinner('O') : setWinner('X');
  };

  const restartGame = () => {
      setWinner('');
      setCurrentPlayer('X');
  }

  return (
    <>
      {winner !== '' ? <p>Winner: {winner}</p> : <p>Player: {currentPlayer}</p>}
      <Board
        current={currentPlayer}
        updateHandler={updatePlayer}
        declareWinner={declareWinner}
        restart={restartGame}
      />
    </>
  );
};

export default Game;
