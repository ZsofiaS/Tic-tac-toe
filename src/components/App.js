import React from 'react';
import styled from 'styled-components';
import Game from './Game';
import GlobalStyle from '../globalStyles';

export const Container = styled.div`
  text-align: center;
  padding: 1rem;
`;

export const Title = styled.h1`
  background: -webkit-linear-gradient(319deg, #118ab2, #06d6a0, #ffd166);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  font-size: 40px;
`;

function App() {
  return (
    <>
      <GlobalStyle />
      <Container>
        <Title>TIC-TAC-TOE</Title>
        <Game />
        <p>Styled components & React Hooks</p>
        <p>
          Made with{' '}
          <span role="img" aria-label="Love">
            ðŸ’•
          </span>{' '}
          by Zsofi
        </p>
      </Container>
    </>
  );
}

export default App;
