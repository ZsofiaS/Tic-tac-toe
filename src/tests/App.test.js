import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App, { Title, Container } from '../components/App';
import Game from '../components/Game';

let appComponent;

beforeEach(() => {
  appComponent = shallow(<App />);
});

describe('renders App component', () => {
  it('renders without crashing', () => {
    shallow(<App />);
  });
  it('should render a h1 element', () => {
    expect(appComponent.find(Title)).toHaveLength(1);
  });
  it('should render a p element', () => {
    expect(appComponent.find('p')).toHaveLength(2);
  });
  it('should render a Game component', () => {
    expect(appComponent.find(Game)).toHaveLength(1);
  });
  it('should render a Container component', () => {
    expect(appComponent.find(Container)).toHaveLength(1);
  });
  it('renders App correctly', () => {
    expect(toJson(appComponent)).toMatchSnapshot();
  });
});
