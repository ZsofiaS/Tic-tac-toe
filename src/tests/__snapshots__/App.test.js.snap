// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component renders App correctly 1`] = `
<Fragment>
  <Memo(l) />
  <styled.div>
    <styled.h1>
      TIC-TAC-TOE
    </styled.h1>
    <Game />
    <p>
      Styled components & React Hooks
    </p>
    <p>
      Made with
       
      <span
        aria-label="Love"
        role="img"
      >
        ðŸ’•
      </span>
       
      by Zsofi
    </p>
  </styled.div>
</Fragment>
`;
